Cython/Compiler/ExprNodes.py

```diff
@@ -13755,6 +13761,21 @@ def analyse_type_annotation(self, env, assigned_value=None):
             warning(annotation.pos, "Unknown type declaration in annotation, ignoring")
         return base_type, arg_type
 
+def WalrusNode(pos, lhs, rhs, **kwds):
+    """
+    An assignment expression
+
+    Defined as a function that constructs a composite node. However, has an interface designed to
+    look like a node so it can easily be replaced if needed.
+    (e.g. for conversion back to Py code.) FIXME!
+    """
+    from .UtilNodes import TempResultFromStatNode, ResultRefNode
+    lhs.walrus_target = True
+    temp = ResultRefNode(pos=pos, expression=rhs)  # assigning expression lets it work out the type
+                        # (but feels a bit like mixing the two uses of ResultRefNode)
+    assignment = Nodes.CascadedAssignmentNode(pos=pos, lhs_list=[lhs, temp], rhs=rhs)
+    return TempResultFromStatNode(result_ref=temp, body=assignment)
```

Comment:
In the current version they're still created at parser-time but in the constructor of `AssignmentExpressionNode`. If that's a problem then it can be postponed.